geode-core/src/main/java/org/apache/geode/distributed/internal/LocatorLoadSnapshot.java

```diff
@@ -50,7 +50,8 @@
 
   private final Map<ServerLocation, String[]> serverGroupMap = new HashMap<>();
```

Comment:
It is possible now, with duplicate `ServerLocation` that `serverGroupMap` could have the wrong groups. It should also use the member ID as its key.

Reply:
done